schemas/ns11_typed_cache_entry.fbs

```diff
@@ -6,34 +6,55 @@ table Double { value:  double; }
 table String { value:  string; }
 table Object { value:  string; } // Python object represented as string
 
-enum ArrayType : byte {
-    ListType = 0,
-    TupleType,
-    SetType
-}   
+// Specific 1-D arrays for simple value types to enable easier serialisation
+// and deserialisation. Also, the numeric types get numpy support for free.
+table ArrayBool    { contents: [bool];    }
+table ArrayLong    { contents: [long];    }
+table ArrayDouble  { contents: [double];  }
+table ArrayString  { contents: [string];  }
+table ArrayObject  { contents: [string];  }
 
-union Value {
+union ValueType {
     Object,
     Bool,
     Long,
     Double,
     String,
     Dict,
-    Array
+    Array,
+    ArrayBool,
+    ArrayLong,
+    ArrayDouble,
+    ArrayString,
+    ArrayObject
 }
 
 table DictMapping {
-        k: Value;
-        v: Value;
+        key: ValueType;
```

Comment:
The only sensible keys I could think of are 
    Bool,
    Long,
    String,
Would it work and make sense to limit to those?